aws/rust-runtime/aws-sigv4/src/http_request/canonical_request.rs

```diff
@@ -338,14 +338,17 @@ impl<'a> fmt::Display for CanonicalRequest<'a> {
         // write out _all_ the headers
         for header in &self.values.signed_headers().headers {
             // a missing header is a bug, so we should panic.
-            let value = &self.headers[&header.0];
+            let values: Vec<&str> = self
+                .headers
+                .get_all(&header.0)
+                .into_iter()
+                .map(|value| {
+                    std::str::from_utf8(value.as_bytes())
+                        .expect("SDK request header values are valid UTF-8")
+                })
+                .collect();
```

Comment:
This change ensures that all values of the header appear in a comma separated list in the `CanonicalHeaders` section on the same row - per the spec.